src/App.tsx

```diff
@@ -1,62 +1,120 @@
+        .then(allPosts => {
+          setPosts(allPosts);
+        })
+        .catch(() => setLoadingPostsError(true))
+        .finally(() => setLoadingPosts(false));
+    }
+  }, [selectedUser]);
+
+  let postsContent;
+
+  if (loadingPosts) {
+    postsContent = <Loader />;
+  } else if (loadingPostsError) {
+    postsContent = (
+      <div
+        className="notification is-danger"
+        data-cy="PostsLoadingError"
+      >
+        Something went wrong!
+      </div>
+    );
+  } else if (selectedUser && posts && posts?.length > 0) {
+    postsContent = (
+      <PostsList
+        userPosts={posts}
+        currentPost={selectedPost}
+        onPostSelected={setSelectedPost}
+      />
+    );
+  } else if (selectedUser && posts && posts.length === 0) {
+    postsContent = (
+      <div className="notification is-warning" data-cy="NoPostsYet">
+        No posts yet
+      </div>
+    );
+  }
+
   return (
     <main className="section">
       <div className="container">
         <div className="tile is-ancestor">
           <div className="tile is-parent">
             <div className="tile is-child box is-success">
               <div className="block">
-                <UserSelector />
+                <UserSelector
+                  users={users}
+                  onUserSelect={setSelectedUser}
+                  userSelected={selectedUser}
+                />
               </div>
 
               <div className="block" data-cy="MainContent">
-                <p data-cy="NoSelectedUser">
-                  No user selected
-                </p>
-
-                <Loader />
-
-                <div
-                  className="notification is-danger"
-                  data-cy="PostsLoadingError"
-                >
-                  Something went wrong!
-                </div>
+                {!selectedUser ? (
+                  <p data-cy="NoSelectedUser">
+                    No user selected
+                  </p>
+                ) : (postsContent)}
 
-                <div className="notification is-warning" data-cy="NoPostsYet">
-                  No posts yet
-                </div>
-
-                <PostsList />
               </div>
             </div>
           </div>
 
-          <div
-            data-cy="Sidebar"
-            className={classNames(
-              'tile',
-              'is-parent',
-              'is-8-desktop',
-              'Sidebar',
-              'Sidebar--open',
-            )}
-          >
-            <div className="tile is-child box is-success ">
-              <PostDetails />
+          {selectedPost && (
+            <div
+              data-cy="Sidebar"
+              className={cn(
+                'tile',
+                'is-parent',
+                'is-8-desktop',
+                'Sidebar',
+                'Sidebar--open',
+              )}
+            >
+              <div className="tile is-child box is-success ">
+                <PostDetails post={selectedPost} />
```

Comment:
hide `<PostDetails post={selectedPost} />` when you choose a new user and check the smoothness of the appearance